src/lexer.c

```diff
@@ -361,10 +397,44 @@ void tryParseExportStatement () {
       ch = commentWhitespace(true);
       while (true) {
         char16_t* startPos = pos;
-        readToWsOrPunctuator(ch);
+
+        bool startWithQuote = isQuote(ch);
+        if (!startWithQuote)
+          ch = readToWsOrPunctuator(ch);
+
         char16_t* endPos = pos;
-        commentWhitespace(true);
+
+        ch = commentWhitespace(true);
+
+        // export { "identifer" as } from
+        // export { "@notid" as } from
+        // export { "spa ce" as } from
+        // export { " space" as } from
+        // export { "space " as } from
+        // export { "not~id" as } from
+        // export { "%notid" as } from
+        // export { "identifer" } from
+        // export { "%notid" } from
+        if (startWithQuote) {
+          char16_t *posQuoteStart = pos;
+
+          stringLiteral(ch);
+
+          char16_t *posQuoteEnd = pos;
+          readToWsOrPunctuator(ch);
+          ch = commentWhitespace(true);
+
+          if (str_eq2(pos, 'a', 's')) {
+            startPos = pos;
+            endPos = pos;
+          } else {
+            startPos = posQuoteStart + 1;
+            endPos = posQuoteEnd;
+          }
+        }
```

Comment:
This entire section should be able to simplify with just the change as below - all three specifier parse cases are just being replaced with this same branching, that's all this PR needs to do:

```
if (!isQuote(ch)) {
      ch = readToWsOrPunctuator(ch);
      endPos = pos;
    }
    // export { mod as "identifer" } from
    // export { mod as "@notid" } from
    // export { mod as "spa ce" } from
    // export { mod as " space" } from
    // export { mod as "space " } from
    // export { mod as "not~id" } from
    // export { mod as "%notid" } from
    else {
      startPos = pos + 1;

      stringLiteral(ch);
      readToWsOrPunctuator(ch);
      endPos = pos - 1;
    }
```

Reply:
Let me try it, I'm afraid maybe there are some special edge cases 🤔

Reply:
I haved tried, if keep `readExportAs` unchanged as main branch. We couldn't do like code above(I removed comment lines):

```
if (!isQuote(ch)) { // branch-1
      ch = readToWsOrPunctuator(ch);
      endPos = pos;
      commentWhitespace(true); // in fact, this line is required on assumption that this snippet is valid
    } else { // branch-2
      startPos = pos + 1;

      stringLiteral(ch);
      readToWsOrPunctuator(ch);
      endPos = pos - 1;
    }
```

for code `export { "@unicode" as identifer } from 'xxx'`, we would extract both `'@unicode'` and `'identifer'`:

1. first, parser meet `"@unicode"`, enter branch-2, startPos represents pos of `@`, after `stringLiteral(ch)` & `readToWsOrPunctuator(ch)`, endPos represents the **space before `as`**, in loop, `ch = readExportAs(startPos, endPos);` would **add export `@unicode`**, but that's **Incorrect** ⚠️
2. then, re-enter of loop, parser will go to **space after `as`**; re-enter loop, parse meet `identifier`, and extract it, this is correct, but we have mistake in step 1.

I have met this case on refactor this entries section, at your reminding, I considered ensuring `readExportAs` always deal with exact whole module name token like:

1. `"@unicode"` in `export { "@unicode" } from 'xxx'`
2. `as identifer` in `export { "@unicode" as identifer } from 'xxx'`
3. `as "@identifer"` in `export { "@unicode" as "@identifer" } from 'xxx'`

So maybe we should do a little bit more changes here on top of main branch, I have pushed my latest change @guybedford 

I guess we are very close to the final minimal changes 😄